source/m2mresourceinstance.cpp

```diff
@@ -337,17 +337,13 @@ void M2MResourceInstance::get_value(uint8_t *&value, uint32_t &value_length)
     }
 }
 
-int M2MResourceInstance::get_value_int()
+int64_t M2MResourceInstance::get_value_int() const
```

Comment:
Null termination is handle in setter. See set_value. Yes, it cannot be 64-bit when use atoi. By the way setter is 64-bit. Maybe this should fix to use atol or atoll.